### PR TITLE
Story security

### PR DESCRIPTION
**Pull request by olivergate**
Login and sessions added

### CODE REVIEW

**Comment by olivergate** [comment]
I will slack the changes to the password file.

File: cms.php [hunk 1]
```diff
@@ -1,6 +1,13 @@
 
 <?php
 require 'pull_data.php';
+session_start();
+if (
+        !isset($_SESSION['admin']) ||
+        $_SESSION['admin'] != 'logged_in'
```
Comment (mporam) [inline cms.php]: this actually isnt long enough to need to be a multi-line if.

File: db_instantiation.php [hunk 2]
```diff
@@ -1,4 +1,11 @@
 <?php
+session_start();
+if (
+    !isset($_SESSION['admin']) ||
+    $_SESSION['admin'] != 'logged_in'
```
Comment (mporam) [inline db_instantiation.php]: neither is this...

File: login_page.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<?php
+session_start();
+if (isset($_GET['login_fail'])) {
+    echo 'LOGIN FAILED';
```
Comment (mporam) [inline login_page.php]: echo here creates invalid HTML, do this check inside your html body tag

File: login_pull.php [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<?php
+require 'password.php';
+session_start();
+if
+(
+    password_verify($_POST['password'], $stored_password) &&
+    password_verify($_POST['username'], $stored_username)
```
Comment (mporam) [inline login_pull.php]: why are you hashing your usernames??? unnecessary

File: login_pull.php [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<?php
+require 'password.php';
+session_start();
+if
+(
+    password_verify($_POST['password'], $stored_password) &&
```
Comment (mporam) [inline login_pull.php:9]: no check if password or username keys exist in the post array

**Review by mporam — CHANGES_REQUESTED** [review]

**Pull request merged**